tests/integrationv2/test_ocsp.py

```diff
@@ -0,0 +1,130 @@
+    port = next(available_ports)
+
+    random_bytes = data_bytes(128)
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        port=port,
+        cipher=cipher,
+        curve=curve,
+        protocol=protocol,
+        insecure=True,
+        data_to_send=random_bytes,
+        enable_client_ocsp=True
+    )
+
+    server_options = ProviderOptions(
+        mode=Provider.ServerMode,
+        port=port,
+        cipher=cipher,
+        curve=curve,
+        protocol=protocol,
+        key=certificate.key,
+        cert=certificate.cert,
+        ocsp_response={
+            "RSA":  TEST_OCSP_DIRECTORY + "ocsp_response.der",
+            "EC":   TEST_OCSP_DIRECTORY + "ocsp_ecdsa_response.der"
+        }.get(certificate.algorithm),
+    )
+
+    kill_marker = None
+
+    if provider == GnuTLS:
+        kill_marker = random_bytes
+
+    server = managed_process(
+        provider,
+        server_options,
+        timeout=30,
+        kill_marker=kill_marker
+    )
+    client = managed_process(S2N, client_options, timeout=30)
+
+    for client_results in client.get_results():
+        client_results.assert_success()
+        assert b"OCSP response received" in client_results.stdout
+
+    for server_results in server.get_results():
+        server_results.assert_success()
+        # Avoid debugging information that sometimes gets inserted after the first character
+        assert random_bytes[1:] in server_results.stdout or random_bytes[1:] in server_results.stderr
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [GnuTLS, OpenSSL])
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", PROTOCOLS, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", OCSP_CERTS, ids=get_parameter_name)
+def test_s2n_server_ocsp_response(managed_process, cipher, provider, curve, protocol, certificate):
+    port = next(available_ports)
+
+    random_bytes = data_bytes(128)
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        port=port,
+        cipher=cipher,
+        curve=curve,
+        protocol=protocol,
+        insecure=True,
+        data_to_send=random_bytes,
+        enable_client_ocsp=True
+    )
+
+    server_options = ProviderOptions(
+        mode=Provider.ServerMode,
+        port=port,
+        cipher=cipher,
+        curve=curve,
+        protocol=protocol,
+        insecure=True,
+        key=certificate.key,
+        cert=certificate.cert,
+        ocsp_response={
+            "RSA":  TEST_OCSP_DIRECTORY + "ocsp_response.der",
+            "EC":   TEST_OCSP_DIRECTORY + "ocsp_ecdsa_response.der"
+        }.get(certificate.algorithm),
+    )
+
+    kill_marker = None
+    if provider == GnuTLS:
+        # the gnutls client hangs for a while after sending. speed up the tests by killing
```

Comment:
Nit: Can you capitalize the beginnings of your sentences in comments 😄

Reply:
Yes sorry: efa04fedb4c48c2ad754e9de3a21e73c77000a24